designdocs/wallettool-subcommands.adoc

```diff
@@ -0,0 +1,579 @@
+
+==== Code Refactoring
+Refactoring the code to use subcommands can address the identified issues effectively by providing a more modular and extensible structure. Here's how:
+
+- **Decoupled Logic with Better Clarity**
+* Each subcommand would encapsulate its specific logic, eliminating the need for large `if` or `switch` structures to determine the preliminary steps for different actions.
+* This separation ensures that commands are isolated and independent, reducing the risk of unintended dependencies or side effects.
+
+- **Action-Specific Fields and Methods**
+* By restructuring the code into subcommands, the data and methods required for a specific action can be encapsulated within that subcommand.
+* This eliminates the problem of ambiguous field usage at the class level and makes the responsibility of each subcommand clearer and more self-contained.
+
+- **Improved Maintainability**
+* Each subcommand acts as a standalone unit with defined responsibilities, making the code easier to understand, maintain, and extend.
+* New commands can be added without affecting existing functionality, avoiding the need to modify a centralized `call()` method.
+
+- **Flexible and Modular Preliminary Checks**
+* Subcommands can handle their own preliminary checks, avoiding a rigid centralized execution flow.
+* This modularity reduces the likelihood of unrelated commands being affected when introducing changes and ensures that responsibilities are distributed cleanly across the system.
+
+- **Extensibility for Future Development**
+* Adding a new command becomes straightforward because it only requires creating a new subcommand and registering it, without altering existing logic.
+* This makes the codebase more adaptable to future requirements.
+
+By leveraging subcommands, the overall design becomes more robust, flexible, and easier to work with, effectively solving the identified challenges in the original implementation.
+
+
+
+
+== wallet-tool: Command
+
+Below shows the proposed format for the subcommand for WalletTool
+
+=== *`wallet-tool`*
+
+*Description*: Print and manipulate wallets. This command is the main command which requires that you specify a subcommand. You may provide a wallet file as an argument. Additionally, you can specify the network with an optional `--net` parameter. If `--net` is not specified, the network will be derived from the wallet.
+
+==== *Options*:
+
+* `--net ++<++NETWORK++>++`: Specifies the network.
+
+==== *Parameter*:
+* `++<++WALLET++_++FILE++>++`: Specifies what wallet file to load and save. If not specified, wallet will be created in default path with a default name. Only subcommand `create` allows you to exempt this parameter.
+
+==== *Usage*:
+
+[source,bash]
+----
+wallet-tool <WALLET_FILE> [--net <NETWORK>] <Subcommand>
```

Comment:
I think the order should be:

wallet-tool  <Subcommand> [options] <WALLET_FILE>